lib/project_types/script/layers/infrastructure/languages/task_runner.rb

```diff
@@ -5,15 +5,35 @@ module Layers
     module Infrastructure
       module Languages
         class TaskRunner
-          TASK_RUNNERS = {
-            "assemblyscript" => AssemblyScriptTaskRunner,
-            "typescript" => TypeScriptTaskRunner,
-            "wasm" => WasmTaskRunner,
-          }
-
           def self.for(ctx, language, script_name)
-            raise Errors::TaskRunnerNotFoundError unless TASK_RUNNERS[language]
-            TASK_RUNNERS[language].new(ctx, script_name)
+            task_runners = {
+              "assemblyscript" => AssemblyScriptTaskRunner,
+              "typescript" => TypeScriptTaskRunner,
+              "wasm" => WasmTaskRunner,
+            }
+
+            raise Errors::TaskRunnerNotFoundError unless task_runners[language]
+            task_runners[language].new(ctx, script_name)
+          end
+
+          def build
+            raise NotImplementedError
+          end
+
+          def dependencies_installed?
+            raise NotImplementedError
+          end
+
+          def install_dependencies
+            raise NotImplementedError
+          end
+
+          def metadata_file_location
+            raise NotImplementedError
+          end
+
+          def library_version(_library_version)
+            raise NotImplementedError
```

Comment:
I've defined the interface here based on the current calls we make to the task_runner externally.